server/service/transport_error.go

```diff
@@ -62,10 +62,12 @@ type existsErrorInterface interface {
 
 // encode error and status header to the client
 func encodeError(ctx context.Context, err error, w http.ResponseWriter) {
+	ctxerr.Handle(ctx, err)
```

Comment:
Instead of in `encodeError`, maybe it makes sense to handle the error (call `ctxerr.Handle`) in `ServerErrorHandler`?
```
		kithttp.ServerErrorHandler(&errorHandler{logger}), // <<<<<
		kithttp.ServerErrorEncoder(encodeError),
```

Reply:
Looking at go-kit code a bit more I think this is a-ok as-is (both error handler and encoder are called in the same place and with the `ctx`, but the docs suggest using `ServerErrorEncoder`...).

Reply:
Yeah I put it there for a couple reasons, I believe the go-kit error handler is for when we actually return an error from the `Endpoint` function, but we don't use this approach, instead we return a response struct that may hold an error or a successful response (and this is determined via the `errorer` interface that this struct can implement). The other thing is that we may want to move away from the go-kit boilerplate/framework at some point, so better not tie ourselves more to it.